shell/common/animator.cc

```diff
@@ -99,7 +99,12 @@ void Animator::BeginFrame(
   FML_DCHECK(producer_continuation_);
   const fml::TimePoint frame_target_time =
       frame_timings_recorder_->GetVsyncTargetTime();
-  dart_frame_deadline_ = frame_target_time.ToEpochDelta();
+  // TODO(moffatman): kind of a hack to deal with pipeline depth 2
+  // It should be based on the latency of the frame
+  // Maybe frame recorder should know the latency?
+  dart_frame_deadline_ =
+      frame_target_time.ToEpochDelta() -
+      ((frame_target_time - frame_timings_recorder_->GetVsyncStartTime()) / 2);
```

Comment:
Can you explain this ? Why does the dart target frame time need to change?

Reply:
Now frame_target_time will be t+2frames instead of t+1 frames. I was seeing NotifyIdle called by one frame and eating too much time out of the other frame (when depth=2) instead of splitting the work up. This fix isn't very good.... it should divide by frame pipeline depth instead of by 2.